source/common/stats/thread_local_store.cc

```diff
@@ -302,6 +331,10 @@ StatType& ThreadLocalStoreImpl::ScopeImpl::safeMakeStat(
 }
 
 Counter& ThreadLocalStoreImpl::ScopeImpl::counter(const std::string& name) {
+  if (parent_.rejectsAll()) {
+    return null_counter_;
+  }
```

Comment:
This is a code smell IMO. Maybe a ThreadLocalStore constructed around a statsmatcher which rejectsall should create a no-op store instead? Or is that simply not possible because statsmatchers can be set after-construction?

Reply:
I thought about this idea for a moment and the main problem with that idea is that we apply the stats matcher after having constructed the store.

In any case, this is a cheap and quick check, and it's now fully tested.